packages/vscode-extension/src/webview/views/PreviewView.tsx

```diff
@@ -154,9 +167,10 @@ function PreviewView() {
           onZoomChanged={onZoomChanged}
         />
       ) : (
-        <div className="missing-device-filler">
-          {devicesNotFound ? <DevicesNotFoundView /> : <VSCodeProgressRing />}
-        </div>
+        <LoadingComponent
+          finishedInitialLoad={finishedInitialLoad}
```

Comment:
Why we're checking for finishedInitialLoad twice in the ternary and in the component?

Reply:
there is more then one reason for rendering the loading component, but not all of them behave the same.